Fix for Issue 89

guytonde opened the pull request:
Issue was around busy waiting and having a timeout of 0 for queue. Changed ticks to wait to portMAX_DELAY to ensure there are scheduling points.

IshDeshpa reviewed (dismissed):
LGTM

Lakshay983 reviewed (dismissed):
LGTM if tested on hardware, but why not pass in how long we want to block for so it can be variable.

guytonde commented:
Changed approach to where the user explicitly specifies the delay amount in delay_ticks (which is of type uint32_t). Made this change for both receive and send. Updated corresponding tests to reflect new method signature.

IshDeshpa commented:
Can you modify the test file and test this behavior on a nucleo?

IshDeshpa reviewed (approved):
Approved.

NathanielDelgado commented on bsp/Src/CAN.c:
```diff
@@ -448,16 +448,12 @@ can_status_t can_recv(CAN_HandleTypeDef* handle, uint16_t id,
-              errQUEUE_EMPTY) {
-            return CAN_EMPTY;
-          }
+
+        // if delay_ticks == portMAX_DELAY thread blocks, 
+        // other values of delay_ticks are delays
```
nitpick but I don't like comments explaining the code. I only like comments that outline or explain why something was done
